mathesar_ui/src/systems/data-explorer/exploration-inspector/ExplorationTab.svelte

```diff
@@ -0,0 +1,148 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import {
+    Collapsible,
+    LabeledInput,
+    TextInput,
+    TextArea,
+    CancelOrProceedButtonPair,
+    Button,
+    Icon,
+  } from '@mathesar-component-library';
+  import { iconDeleteMajor } from '@mathesar/icons';
+  import type { QueryInstance } from '@mathesar/api/queries';
+  import { queries, putQuery, deleteQuery } from '@mathesar/stores/queries';
+  import { confirmDelete } from '@mathesar/stores/confirmation';
+  import { getAvailableName } from '@mathesar/utils/db';
+  import Form from '@mathesar/components/Form.svelte';
+  import FormField from '@mathesar/components/FormField.svelte';
+  import { toast } from '@mathesar/stores/toast';
+  import type QueryRunner from '../QueryRunner';
+  import QueryManager from '../QueryManager';
+
+  const dispatch = createEventDispatcher();
+
+  export let queryHandler: QueryRunner | QueryManager;
+  export let name: string | undefined;
+  export let description: string | undefined;
+
+  $: ({ query } = queryHandler);
+  $: hasManager = queryHandler instanceof QueryManager;
+
+  let hasChanges = false;
+
+  function setHasChangesToTrue() {
+    hasChanges = true;
+  }
+
+  function handleNameChange(e: Event) {
+    const target = e.target as HTMLInputElement;
+    if (target.value.trim() === '') {
+      target.value =
+        $query.name ??
+        getAvailableName(
+          'New_Exploration',
+          new Set([...$queries.data.values()].map((q) => q.name)),
+        );
+    }
+    name = target.value;
+    if ('update' in queryHandler) {
+      void queryHandler.update((q) => q.withName(target.value));
+    }
+  }
+
+  function handleDescriptionChange(e: Event) {
+    const target = e.target as HTMLInputElement;
+    description = target.value.trim();
+    if ('update' in queryHandler) {
+      void queryHandler.update((q) => q.withDescription(description ?? ''));
+    }
+  }
+
+  function handleCancel() {
+    name = $query.name;
+    description = $query.description;
+    hasChanges = false;
+  }
+
+  async function handleSave() {
+    try {
+      const updatedQuery = $query
+        .withName(name)
+        .model.withDescription(description).model;
+      // TODO: Write better utility methods to identify saved instances
+      await putQuery(updatedQuery.toJSON() as QueryInstance);
+      query.set(updatedQuery);
+    } catch (err) {
+      const message =
+        err instanceof Error ? err.message : 'Unable to save Exploration.';
```

Comment:
_(No changes needed)_

I have been using our `getErrorMessage` utility function for this. I like the way you've implemented a fallback message here, and it strikes me that we could add an optional parameter to `getErrorMessage` to accept a fallback message like this that would get used before the `JSON.stringify` step.

I'm mostly just thinking out loud here and wanting to make sure you're aware of the `getErrorMessage` function. It might be nice for us to eventually use a consistent pattern.